public/js/src/articleListModel.js

```diff
@@ -0,0 +1,29 @@
+(function(exports) {
+  function ArticleList() {
+    this.list = []
+  }
+  ArticleList.prototype.listAll = function() {
+    return this.list
+  }
+  ArticleList.prototype.add = function(headline) {
+    let article = new Article(headline)
+    this.list.push(article)
+    article.id = this.list.indexOf(article)
+  }
+
+  ArticleList.prototype.getData = function() {
+    let self = this
+    fetch('http://news-summary-api.herokuapp.com/guardian?apiRequestUrl=http://content.guardianapis.com/travel')
+      .then(response => response.json())
+      .then(data => {
+        let results = data.response.results;
+        results.forEach((item, i) => {
+          self.add(item.webTitle)
+          self.list[i].url = item.apiUrl
+        });
+        articleController.insertHTML()
```

Comment:
This is not the best place to have that, as it makes you ArticleList and ArticleController classes very coupled. Instead, you could have something like:
in ArticleList:
```js
ArticleList.prototype.getData = function(callback) {
    let self = this
    fetch('http://news-summary-api.herokuapp.com/guardian?apiRequestUrl=http://content.guardianapis.com/travel')
      .then(response => response.json())
      .then(data => {
        let results = data.response.results;
        results.forEach((item, i) => {
          self.add(item.webTitle)
          self.list[i].url = item.apiUrl
        });
      callback()
      })
  }
```

And in articleController's constructor:
```js
let self = this
this.articleList.getData(self.insertHTML)
```

This way you know `insertHTML` will run after the results have come back, but the ArticleList class doesn't need to know about functionalities of the ArticleController.